components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksUseCases.kt

```diff
@@ -218,4 +230,16 @@ class AppLinksUseCases(
         val marketplaceIntent: Intent? = null,
         val resolveInfo: ResolveInfo? = null
     )
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal data class AppLinkRedirectCache(
+        var cacheTimeStamp: Long = 0,
+        var cachedUrlHash: Int = 0,
+        var cachedAppLinkRedirect: AppLinkRedirect = AppLinkRedirect(null, null, null)
+    )
+
+    companion object {
+        @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+        internal var redirectCache = AppLinkRedirectCache()
```

Comment:
Instead of having default parameters for the cache type, just make this field nullable.

Reply:
> Instead of having default parameters for the cache type, just make this field nullable.

I was on the fence about this.  Do you mean making `redirectCache` nullable but then when returning, do `return redirectCache?.cachedAppLinkRedirect ?: AppLinkRedirectCache()`

I like how clean it is right now though since we don't have to check what's in the cache when returning.

Reply:
Since you don't have to return the cache, I think you can just make this change:
```kt
if (urlHash == redirectCache?.cachedUrlHash && ...) {
```

Reply:
Oh doh, that makes perfect sense.  I'll update.  Thanks 😅

Reply:
Oh I remembered why now.  Since `redirectCache` is mutable.  I can't be sure that after the check it can be set to null again.  So guarding the return inside doesn't work.  I can't change it to a Nullable.

Reply:
Updated.  Thanks for the suggestion.